Assets/Creatures/api/abilities/infusion/CoreInfusionAbilityEvent.cs

```diff
@@ -13,5 +14,15 @@ public override AbilityExecutionResult ExecuteAbility(IPlayable playable)
         {
             return playable.CoreInfusionAbility(this);
         }
+        
+        public override void Subscribe(IStateNotifier notifier, CooldownState state)
+        {
+            notifier.Subscribe(state, this);
```

Comment:
is there no way for them to easily share these methods that are basically the same? I see the `this` being the problematic part, so maybe it's the way it is 🤷

Reply:
Talked in chat about this, but you need this because the caller doesn't know what type of `AbilityEvent` this is and we are essentially delegating the problem of determining identity to the implementations